Client/U2FExtensions.swift

```diff
@@ -465,6 +576,57 @@ class U2FExtensions: NSObject {
         }) }
     }
     
+    private func presentInteractWithKeyModal() {
+        observeSessionStateUpdates = false
+        observeKeyStateUpdates = false
+        defer {
+            observeSessionStateUpdates = true
+            observeKeyStateUpdates = true
+        }
+        
+        guard let fido2Service = YubiKitManager.shared.keySession.fido2Service else {
+            return
+        }
+        
+        guard let u2fService = YubiKitManager.shared.keySession.u2fService else {
+            return
+        }
+        
+        if fido2Service.keyState == .touchKey || u2fService.keyState == .YKFKeyU2FServiceKeyStateTouchKey {
+            popup.showWithType(showType: .flyUp)
+            return
+        }
+        popup.dismissWithType(dismissType: .flyDown)
+    }
+    
+    private func handlePinVerificationRequired(completion: @escaping (Bool) -> Void ) {
+        ensureMainThread {
+            let alert = UIAlertController.userTextInputAlert(title: Strings.pinTitle, message: Strings.pinLabel, placeholder: Strings.pinPlaceholder) {
+                pin, _ in
+                if let pin = pin, !pin.isEmpty {
+                    guard let fido2Service = YubiKitManager.shared.keySession.fido2Service else {
+                        completion(true)
+                        return
+                    }
+                    guard let verifyPinRequest = YKFKeyFIDO2VerifyPinRequest(pin: pin) else {
+                        completion(true)
+                        return
+                    }
+
+                    fido2Service.execute(verifyPinRequest) { (error) in
+                        guard error == nil else {
+                            completion(true)
+                            return
+                        }
+                        completion(false)
```

Comment:
After calling `completion(false)`, do you need a `return` statement or is showing the alert what you intended? It looks intentional because the same logic is used in `handlePinVerificationRequired`

Reply:
I can add a `return` there, but I don't think it matters. Its just the end of the callback and it will return after invoking `completion(false)`

Reply:
Done!